ompi/communicator/comm_cid.c

```diff
@@ -858,16 +859,36 @@ static int ompi_comm_allreduce_pmix_reduce_complete (ompi_comm_request_t *reques
     opal_dss.unload(&sbuf, (void**)&info.data.bo.bytes, &info.data.bo.size);
     OBJ_DESTRUCT(&sbuf);
 
-    if (cid_context->send_first) {
-        (void)asprintf(&info.key, "%s:%s:send:%d", cid_context->port_string, cid_context->pmix_tag,
-                       cid_context->iter);
-        (void)asprintf(&pdat.value.key, "%s:%s:recv:%d", cid_context->port_string, cid_context->pmix_tag,
-                       cid_context->iter);
+    bytesWritten = asprintf(&info.key,
+                            cid_context->send_first ? "%s:%s:send:%d"
+                                                    : "%s:%s:recv:%d",
+                            cid_context->port_string,
+                            cid_context->pmix_tag,
+                            cid_context->iter);
+
+    if (bytesWritten == -1) {
+        fprintf (stderr, "writing info.key failed\n");
     } else {
-        (void)asprintf(&info.key, "%s:%s:recv:%d", cid_context->port_string, cid_context->pmix_tag,
-                       cid_context->iter);
-        (void)asprintf(&pdat.value.key, "%s:%s:send:%d", cid_context->port_string, cid_context->pmix_tag,
-                       cid_context->iter);
+        bytesWritten = asprintf(&pdat.value.key,
+                                cid_context->send_first ? "%s:%s:recv:%d"
+                                                        : "%s:%s:send:%d",
+                                cid_context->port_string,
+                                cid_context->pmix_tag,
+                                cid_context->iter);
+
+        if (bytesWritten == -1) {
+            fprintf (stderr, "writing pdat.value.key failed\n");
+        }
+    }
+
+    if (bytesWritten == -1) {
+        // write with separate calls,
+        // just in case the args are the cause of failure
+        fprintf (stderr, "send first: %d\n", cid_context->send_first);
+        fprintf (stderr, "port string: %s\n", cid_context->port_string);
+        fprintf (stderr, "pmix tag: %s\n", cid_context->pmix_tag);
+        fprintf (stderr, "iter: %d\n", cid_context->iter);
+        return OMPI_ERR_FATAL;
```

Comment:
You can probably return `OMPI_ERR_OUT_OF_RESOURCE` here; it should be treated as a fatal error by the caller.